### PR TITLE
bugfix(spacebindingrequest): allow deletion of failed SpaceBindingRequest

### PR DESCRIPTION
**Pull request by mfrancisc**
This fixes the issue with SpaceBindingRequest not being deleted when there is already an existing SpaceBinding object not owned by the SBR resource.

Jira: https://issues.redhat.com/browse/ASC-442

### CODE REVIEW

File: controllers/spacebindingrequest/spacebindingrequest_controller.go [hunk 1]
```diff
@@ -260,6 +266,18 @@ func (r *Reconciler) ensureSpaceBinding(ctx context.Context, memberCluster clust
...
+// sbrOwnsSpaceBinding check if the SpaceBinding was created by the given SpaceBindingRequest resource
+// returns true if it was created by the SBR
+// returns false if it wasn't created by the SBR
+func sbrOwnsSpaceBinding(spaceBinding *toolchainv1alpha1.SpaceBinding, spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest) bool {
+	sbrLabel := spaceBinding.Labels[toolchainv1alpha1.SpaceBindingRequestLabelKey]
+	sbrNamespaceLabel := spaceBinding.Labels[toolchainv1alpha1.SpaceBindingRequestNamespaceLabelKey]
+	if sbrLabel != spaceBindingRequest.GetName() || sbrNamespaceLabel != spaceBindingRequest.GetNamespace() {
+		return false
+	}
+	return true
```
Comment (sadlerap) [inline controllers/spacebindingrequest/spacebindingrequest_controller.go:278]:
```suggestion
	return sbrLabel == spaceBindingRequest.GetName() && sbrNamespaceLabel == spaceBindingRequest.GetNamespace()
```

**Review by sadlerap — COMMENTED** [review]
One small nit, otherwise LGTM

File: controllers/spacebindingrequest/spacebindingrequest_controller.go (hunk 1 above)
Comment (mfrancisc) [reply controllers/spacebindingrequest/spacebindingrequest_controller.go:278]: indeed this is shorter/nicer! thanks a lot!

**Review by sadlerap — APPROVED** [review]

**Pull request merged**